# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Forecasting  
I chose forecasting because FlyRank’s search data shows clear daily fluctuations in client interactions. Anticipating these patterns can help allocate support resources more efficiently. This lane is practical, connects directly to the dataset, and has immediate decision‑making value.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

- **Decision improved:** How many staff and technical resources should be scheduled each day to handle client interactions.  
- **Who acts on it:** Operations managers and support teams who plan staffing levels and system capacity.  
- **Cost of a wrong call:**  
- **Underestimate:** Clients face delays, poor service, and higher churn risk.  
- **Overestimate:** Wasted staffing costs and idle resources that reduce efficiency.  


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv(r"C:\Users\mokga\Projects\ml-internship-leonard\data\raw\content_refresh_anonymized.csv")


# Show basic stats for the 'interactions' column (replace with the actual column name if different)
df['engagement_rate'].describe()


count    30000.000000
mean         2.534520
std          8.310096
min          0.000000
25%          0.000000
50%          0.000000
75%          1.350000
max        100.000000
Name: engagement_rate, dtype: float64

# Research Question

We want to understand what features predict whether content refresh events lead to user engagement. 
The `engagement_rate` column shows that most content has zero engagement, with a small fraction achieving high values. 
This skew suggests a classification framing: engaged (rate > 0) vs not engaged (rate = 0). 
Our question: *Which content features (CTR, avg_position, scroll_rate, competition_level, etc.) best predict engagement?*


In [8]:
# Create a binary target for engagement
df['engaged'] = (df['engagement_rate'] > 0).astype(int)

# Check balance
df['engaged'].value_counts(normalize=True)


engaged
0    0.720967
1    0.279033
Name: proportion, dtype: float64

# Research Question

The dataset shows that 72% of content refresh events have zero engagement, while 28% show positive engagement. 
This imbalance suggests a classification problem: predicting whether a refresh will lead to engagement (`engaged = 1`) or not (`engaged = 0`). 

**Research Question:**  
Which content features (CTR, avg_position, scroll_rate, competition_level, word_count, etc.) best predict whether a content refresh event will engage users?


In [9]:
# Binary target already created
df['engaged'] = (df['engagement_rate'] > 0).astype(int)

# Check proportions
df['engaged'].value_counts(normalize=True).rename("proportion")


engaged
0    0.720967
1    0.279033
Name: proportion, dtype: float64

In [10]:
df[['search_volume','impressions_90d','clicks_90d']].describe()


,search_volume,impressions_90d,clicks_90d
count,27532.000000,30000.000000,30000.000000
mean,158.882391,5200.366300,16.097333
std,1518.270825,16838.019547,75.076958
min,0.000000,1.000000,0.000000
25%,0.000000,81.000000,0.000000
50%,10.000000,731.000000,1.000000
75%,20.000000,3615.250000,7.000000
max,74000.000000,517715.000000,4178.000000


From the dataset:

- **Average search volume:** ~159  
- **Median impressions (90 days):** ~731  
- **Maximum clicks (90 days):** ~4,178  

These numbers show large variability in content performance. Forecasting these metrics can help anticipate demand and allocate resources more effectively. A wrong forecast could mean missing a surge in traffic or wasting effort on content that won’t perform.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**Careful words: what I can and can’t claim**

- I can claim that the dataset shows clear variability in search volume, impressions, and clicks.  
- I can claim that forecasting these metrics could help managers anticipate demand and allocate resources more effectively.  
- I cannot claim that my forecast will be perfectly accurate or guarantee business outcomes.  
- I cannot claim causation (e.g., that higher impressions directly cause more revenue) — only correlation or observed patterns.  
- I cannot claim that the dataset represents all clients or all possible scenarios; it is a sample with its own limits.  


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

# ✅ Self‑check

- Dataset loads correctly from `data/raw/content_refresh_anonymized.csv` ✔  
- Research question framed around engagement (`engagement_rate`) ✔  
- Binary target column `engaged` created ✔  
- Class proportions confirmed (72% no engagement, 28% engagement) ✔  
- Notebook runs top to bottom with no errors ✔  
- No client names, URLs, or private queries included ✔  
- Claims use careful words: observed, measured, directional, decision‑support ✔  
- Committed under `work/notebooks/w01_research_question.ipynb` ✔  
Ready to submit repo URL for ML‑01 completion credit.
